bindings/csharp/Ckzg.Bindings/Ckzg.cs

```diff
@@ -1,115 +1,231 @@
-    public const int BytesPerBlob = BytesPerFieldElement * 4096;
+    public const int FieldElementsPerBlob = 4096;
+    public const int BytesPerBlob = BytesPerFieldElement * FieldElementsPerBlob;
     public const int BytesPerCommitment = 48;
     public const int BytesPerProof = 48;
+    public const int BytesScalar = 32;
 
-    public enum Ret
+    private static readonly ThreadLocal<SHA256> _sha256 = new(SHA256.Create);
+
+    /// <summary>
+    /// </summary>
+    /// <param name="commitment"></param>
+    /// <param name="hashBuffer"></param>
+    /// <returns></returns>
+    /// <exception cref="ArgumentException"></exception>
+    public static bool TryComputeCommitmentV1(ReadOnlySpan<byte> commitment, Span<byte> hashBuffer)
     {
-        Ok,
-        BadArgs,
-        Error,
-        Malloc
-    }
+        const int bytesPerHash = 32;
+        const int kzgBlobHashVersionV1 = 1;
+
+        if (commitment.Length != BytesPerCommitment) throw new ArgumentException("Incorrect size", nameof(commitment));
+        if (hashBuffer.Length != bytesPerHash) throw new ArgumentException("Incorrect size", nameof(commitment));
+        if (_sha256.Value!.TryComputeHash(commitment, hashBuffer, out _))
+        {
+            hashBuffer[0] = kzgBlobHashVersionV1;
+            return true;
+        }
 
-    static Ckzg() => AssemblyLoadContext.Default.ResolvingUnmanagedDll += (assembly, path) => NativeLibrary.Load($"runtimes/{(
-            RuntimeInformation.IsOSPlatform(OSPlatform.Linux) ? "linux" :
-            RuntimeInformation.IsOSPlatform(OSPlatform.OSX) ? "osx" :
-            RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? "win" : "")}-{RuntimeInformation.ProcessArchitecture switch
-            {
-                Architecture.X64 => "x64",
-                Architecture.Arm64 => "arm64",
-                _ => ""
-            }}/native/{path}.{(RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? "dll" : "so")}");
+        return false;
+    }
 
     /// <summary>
-    /// Load trusted setup settings from file
+    ///     Load trusted setup settings from file
     /// </summary>
     /// <param name="filename">Settings file path</param>
     /// <returns>Trusted setup settings as a pointer or <c>0</c> in case of failure</returns>
-    [DllImport("ckzg", EntryPoint = "load_trusted_setup_wrap")]
-    public static extern IntPtr LoadTrustedSetup(string filename);
+    public static IntPtr LoadTrustedSetup(string filename)
+    {
+        if (!File.Exists(filename)) throw new ArgumentException("Trusted setup file does not exist", nameof(filename));
+
+        IntPtr ckzgSetup = InternalLoadTrustedSetup(filename);
+
+        if (ckzgSetup == IntPtr.Zero) throw new InvalidOperationException("Unable to load trusted setup");
+        return ckzgSetup;
+    }
 
     /// <summary>
-    /// Frees memory allocated for trusted setup settings
+    ///     Frees memory allocated for trusted setup settings
```

Comment:
Missing punctuation here.

Reply:
Could you clarify

Reply:
Ah yeah. Just add a period to the end, like:

```
///     Frees memory allocated for trusted setup settings.
```